---
title: 'How does the math for DeSeq2 work (library normalization)?'
date: 2024-07-21
permalink: /posts/2013/08/blog-post-2/
tags:
  - Math
  - Learning
  - Deseq2
---

For this blog post I took inspiration from [DeSeq2 Library Normalization](https://www.youtube.com/watch?v=UFB993xufUU&ab_channel=StatQuestwithJoshStarmer). I just could not find the math for DeSeq2 written out simply thus I wrote this. 

Now for the content. Why do we even use DeSeq2?

DeSeq2 is very useful due to adjusting for differences in library sizes (raw counts) due to differing sequencing depth. For example if my accessory venom gland has 700 reads but my main venom gland has 2,000 DeSeq2 is able to accomodate that. 

Another reason Deseq2 is very useful is that it can adjust for differential expression among different tissue types such as my accessory vs main venom gland. This can be because certain genes might naturally be more expressed in accessory compared to the main venom gland thus normalizing it and getting an accurate differential expression is important.


Step 1
======
DeSeq2 takes the raw library counts and takes the log of each sample. 

Here we will start out with the raw count table and transform it by putting each number of these numbers through log(e). 

Both tables were made using LaTeX

<br/><img src='/images/DeSeq2_table.png'>


When all the numbers are put through latex it leads to the table below

<br/> <img src='/images/DeSeq2_logtable.png'>

Step 2
======




Step 3
======


