---
title: 'How does the math for DeSeq2 work (library normalization)?'
date: 2024-07-21
permalink: /posts/2013/08/blog-post-2/
tags:
  - Math
  - Learning
  - Deseq2
---

For this blog post I took inspiration from [DeSeq2 Library Normalization](https://www.youtube.com/watch?v=UFB993xufUU&ab_channel=StatQuestwithJoshStarmer). I just could not find the math for DeSeq2 written out simply thus I wrote this. 

Now for the content. Why do we even use DeSeq2?

DeSeq2 is very useful due to adjusting for differences in library sizes (raw counts) due to differing sequencing depth. For example if my accessory venom gland has 700 reads but my main venom gland has 2,000 DeSeq2 is able to accomodate that. 

Another reason Deseq2 is very useful is that it can adjust for differential expression among different tissue types such as my accessory vs main venom gland. This can be because certain genes might naturally be more expressed in accessory compared to the main venom gland thus normalizing it and getting an accurate differential expression is important.


Step 1
======
DeSeq2 takes the raw library counts and takes the log of each sample. 

Here we will start out with the raw count table and transform it by putting each number of these numbers through log(e). 

| Gene | Sample AVG raw count #1 | Sample AVG raw count # 2 | Sample MVG raw count # 3 | Sample MVG raw count #4
| :-: | :-: | :-: | :-: | :-:
| Gene #1 |10|20|9|0
| Gene #2 |20|13|15|8
| Gene #3 |5|6|19|8
| Gene #4 |10|8|7|9


When these raw counts are put through log(e) it leads to this:

| Gene | Sample AVG log(e) #1 | Sample AVG log(e) # 2 | Sample MVG log(e) # 3 | Sample MVG log(e) #4
| :-: | :-: | :-: | :-: | :-:
| Gene #1 |2.3|8.686|3.909|-Inf
| Gene #2 |8.686|5.646|6.514|3.474
| Gene #3 |2.171|2.606|8.252|3.474
| Gene #4 |2.3|3.474|3.04|3.909


Step 2
======

We then take the geometric averages (averages of log values) of each row (AVG Gene #1,#2;MVG Gene #1,#2) from log(e) table from Step 1.  

| Gene | Averaged log |
| :-: | :-: | :-: |
| Gene #1 |-Inf|
| Gene #2 |6.08|
| Gene #3 |4.125|
| Gene #4 |3.18075|

DeSeq2 will remove the -Inf numbers because it means that it is only expressed in one type of the sample such as it only being expressed in MVG vs AVG.

| Gene | Averaged log |
| :-: | :-: | :-: |
| Gene #2 |6.08|
| Gene #3 |4.125|
| Gene #4 |3.18075|


Step 3
======

DeSeq2 will then substract the individual log value from step 1 to the geometric average.

| Gene | Sample AVG #1 | Sample AVG # 2 | Sample MVG # 3 | Sample MVG #4
| :-: | :-: | :-: | :-: | :-:
| Gene #2 |2.606|0.434|0.434|-2.606
| Gene #3 |-1.954|-1.519|4.127|-0.651
| Gene #4 |-0.88075|0.29325|-0.14075|0.72825

This is quantifying how close to the average these genes are.

Step 4
======

We then have to obtain all the medians of the samples across all the genes from the substracted values from step 3! 

| Gene | Sample AVG #1 Average | Sample AVG # 2 Average| Sample MVG # 3 Average | Sample MVG #4 Average
| :-: | :-: | :-: | :-: | :-:
|Averages|-0.07625|-0.26391667|1.4734|-0.8429

Step 5
======

We then raise all these median numbers to e in order to normalize it and get scaling factors (SF).

| Gene | Sample AVG #1 SF | Sample AVG # 2 (SF)| Sample MVG # 3 (SF)| Sample MVG #4 (SF)
| :-: | :-: | :-: | :-: | :-:
|SF|0.9265|0.768037|4.36404|0.4304|

Step 6
======
We then take these scaling factors and take the original counts and divide it by them.

| Gene | Sample AVG scaled #1 | Sample AVG scaled # 2 | Sample MVG scaled # 3 | Sample MVG scaled #4
| :-: | :-: | :-: | :-: | :-:
| Gene #1 |10.79|26.04|2.062|0
| Gene #2 |21.58|16.926|3.437|18.587
| Gene #3 |5.396|7.812|4.353|18.587
| Gene #4 |10.79|10.416|1.604|20.910

This is the new scaled factors in which some samples such as Sample #3 is downscaled while the rest is upscaled in order to normalize these gene expressions. This is the capabilities of DeSeq2 in terms of determining true differential expression. Thank you again to StatQuest for explaining this to me so I can explain it in text!




